clash-prelude/src/Clash/Annotations/Primitive.hs

```diff
@@ -241,37 +273,19 @@ data Primitive
   -- ^ Description of a primitive for a given 'HDL's as an inline 'String'
   deriving (Show, Read, Data, Generic, NFData, Hashable)
 
--- | Guard primitive functions. This will help Clash generate better error
--- messages. You can annotate a function like:
---
--- @
--- {\-\# ANN f dontTranslate \#-\}
--- @
---
--- or
---
--- @
--- {\-\# ANN f hasBlackBox \#-\}
--- @
+-- | Primitive guard to mark a value as either not translatable or as having a
+-- blackbox with an optional extra warning. Helps Clash generate better error
+-- messages.
 --
--- or
---
--- @
--- {\-\# ANN f (warnNonSynthesizable "Tread carefully, user!") \#-\}
--- @
---
--- or
---
--- @
--- {\-\# ANN f (warnAlways "Tread carefully, user!") \#-\}
--- @
+-- For use, see `dontTranslate`, `hasBlackBox`, `warnNonSynthesizable` and
```

Comment:
If you use single quotes to reference functions it will get properly linked in the documentation. E.g.,

```
'dontTranslate'
```

Reply:
Sure, but it looks like they do the same thing: [https://haskell-haddock.readthedocs.io/en/latest/markup.html#hyperlinked-identifiers](https://haskell-haddock.readthedocs.io/en/latest/markup.html#hyperlinked-identifiers)

Reply:
Ah, I thought it didn't. In that case it's fine!

Reply:
Changed it anyway while I was at it.